presto-main/src/main/java/io/prestosql/server/PrefixObjectNameGeneratorModule.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import org.weakref.jmx.ObjectNameBuilder;
+import org.weakref.jmx.ObjectNameGenerator;
+
+import java.util.Map;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.util.Objects.requireNonNull;
+
+public class PrefixObjectNameGeneratorModule
+        implements Module
+{
+    private final String basePackageName;
+
+    public PrefixObjectNameGeneratorModule(String basePackageName)
+    {
+        this.basePackageName = requireNonNull(basePackageName, "basePackageName is null");
+    }
+
+    @Override
+    public void configure(Binder binder)
+    {
+        configBinder(binder).bindConfig(JmxNamingConfig.class);
+    }
+
+    @Provides
+    ObjectNameGenerator createPrefixObjectNameGenerator(JmxNamingConfig jmxNamingConfig)
+    {
+        return new PrefixObjectNameGenerator(basePackageName, jmxNamingConfig.getDomainBase());
+    }
+
+    public static final class PrefixObjectNameGenerator
+            implements ObjectNameGenerator
+    {
+        private final String basePackageName;
+        private final String domainBase;
+
+        public PrefixObjectNameGenerator(String basePackageName, String domainBase)
```

Comment:
Maybe call the arguments "prefix" and "replacement" to clarify what they are used for.